ark/utils/notebooks_test.py

```diff
@@ -1,125 +1,567 @@
+    """
+    POST_CLUSTERING: pathlib.Path = templates_dir / "4_Post_Clustering.ipynb"
+    with testbook(POST_CLUSTERING, timeout=6000, execute=False) as nb_context_manager:
+        yield nb_context_manager, base_dir_generator / "nb4"
+
+
+@pytest.fixture(scope="class")
+def nbfib_seg_context(templates_dir, base_dir_generator) -> Iterator[ContextManager]:
+    """
+    Creates a testbook context manager for the fiber segmentation notebook.
+
+    Args:
+        templates_dir (pytest.Fixture): The fixture which yields the directory of the notebook
+            templates
+        base_dir_generator (pytest.Fixture): The fixture which yields the temporary directory
+            to store all notebook input / output.
+
+    Yields:
+        Iterator[ContextManager]: The testbook context manager which will get cleaned up
+            afterwords.
+    """
+    EXAMPLE_FIBER_SEGMENTATION: pathlib.Path = templates_dir / "example_fiber_segmentation.ipynb"
+    with testbook(EXAMPLE_FIBER_SEGMENTATION, timeout=6000, execute=False) as nb_context_manager:
+        yield nb_context_manager, base_dir_generator / "efs"
+
+
+@pytest.mark.order(0)
+class Test_1_Segment_Image_Data:
+    """
+    Tests Notebook 1 - Segment Image Data for completion.
+    NOTE: When modifying the tests, make sure the test are in the
+    same order as the tagged cells in the notebook.
+    """
+    @pytest.fixture(autouse=True, scope="function")
+    def _setup(self, nb1_context):
+        """
+        Sets up necessary data and paths to run the notebooks.
+        """
+        self.tb: testbook = nb1_context[0]
+        self.base_dir: pathlib.Path = nb1_context[1]
+
+    def test_imports(self):
+        self.tb.execute_cell("import")
+
+    def test_base_dir(self):
+        base_dir_inject = f"""
+            base_dir = "{self.base_dir}"
+        """
+        self.tb.inject(base_dir_inject, "base_dir")
+
+    def test_file_path(self):
+        self.tb.execute_cell("file_path")
+
+    def test_create_dirs(self):
+        self.tb.execute_cell("create_dirs")
+
+    def test_ex_data_download(self):
+        self.tb.execute_cell("ex_data_download")
+
+    def test_validate_path(self):
+        self.tb.execute_cell("validate_path")
+
+    def test_load_fovs(self):
+        load_fovs_inject = """
+            fovs = ["fov0", "fov1"]
+        """
+        self.tb.inject(load_fovs_inject, "load_fovs")
+
+    def test_nuc_mem_set(self):
+        self.tb.execute_cell("nuc_mem_set")
+
+    def test_gen_input(self):
+        self.tb.execute_cell("gen_input")
+
+    def test_seg_scale_set(self):
+        self.tb.execute_cell("seg_scale_set")
+
+    def test_create_output(self):
+        # Get Deepcell Output Dir from the notebook
+        deepcell_output_dir = self.tb.ref("deepcell_output_dir")
+        fovs = self.tb.ref("fovs")
+        # Generate the sample feature_0, feature_1 tiffs
+        notebooks_test_utils.generate_sample_feature_tifs(fovs, deepcell_output_dir, (1024, 1024))
```

Comment:
This is one of the only substantial injections. Is this to avoid the necessity of downloading the example dataset, or sending the data to the deepcell server?

Reply:
Yeah this is for avoiding sending anything to the Deepcell service directly.